pkg/shim/shim.go

```diff
@@ -449,6 +454,13 @@ func serve(ctx context.Context, server *ttrpc.Server, signals chan os.Signal, sh
 			log.G(ctx).WithError(err).Fatal("containerd-shim: ttrpc server failure")
 		}
 	}()
+
+	if debugFlag {
```

Comment:
How is `debug-socket` flag getting passed down to the shim call?

Reply:
I have analyzed the code. 
@mxpv  Only the `-debug` flag is needed to enable pprof.
@dmcgowan  The `-debug-socket` is not necessary. If `-debug-socket` is not passed, shim calls `os.NewFile(4, 'socket')` to obtain the fd passed by containerd.
@henry118  Is my understanding correct?

Reply:
The unix socket address is generated by `sha256.Sum256`.
```
[root@devnode1 containerd]# ctr c ls
CONTAINER    IMAGE                             RUNTIME
mydemo       docker.io/library/nginx:latest    io.containerd.runc.v2
[root@devnode1 containerd]# ls /run/containerd/s/
0b6efc3510d8ebcb7d27d04dfe4444511fcb6a1fb95411e15b50888cdec86dc3
12bede06714e38de587a8a27a83b967a4a5263ff48c6186a4ec274927a8bc692
[root@devnode1 containerd]# echo -n /run/containerd/containerd.sock/default/mydemo | sha256sum
0b6efc3510d8ebcb7d27d04dfe4444511fcb6a1fb95411e15b50888cdec86dc3  -
[root@devnode1 containerd]# echo -n /run/containerd/containerd.sock/default/mydemo/debug | sha256sum
12bede06714e38de587a8a27a83b967a4a5263ff48c6186a4ec274927a8bc692  -
```

Reply:
Yes the `debug-socket` is not used or passed down in regular case, where the shim is spawned by containerd.

I created this cmd option for two reasons:
1. By following the existing pattern in shim where the `socket` cmd option is optionally passed in and used as the socket for all shim requests. Similarly when shim is spawned by containerd the `socket` option is not used, and the socket addr is generated in the format of `containerd-sock/ns/container-id`;
2. IMO both `socket` and `debug-socket` would be useful when debugging the shim standalone (manual launch instead of spawned by containerd). In this case a user can specify a well-known socket addr.